service/worker/deployment/deployment_workflow.go

```diff
@@ -191,7 +151,70 @@ func (d *DeploymentWorkflowRunner) run() error {
 
 	d.logger.Debug("Deployment doing continue-as-new")
 	return workflow.NewContinueAsNewError(d.ctx, DeploymentWorkflow, d.DeploymentWorkflowArgs)
+}
+
+func (d *DeploymentWorkflowRunner) validateRegisterWorker(args *deploymentspb.RegisterWorkerInDeploymentArgs) error {
+	if _, ok := d.DeploymentLocalState.TaskQueueFamilies[args.TaskQueueName].GetTaskQueues()[int32(args.TaskQueueType)]; ok {
+		return errAlreadyExists
+	}
+	return nil
+}
+
+func (d *DeploymentWorkflowRunner) handleRegisterWorker(ctx workflow.Context, args *deploymentspb.RegisterWorkerInDeploymentArgs) error {
+	// Note: use ctx in here (provided by update) instead of d.ctx
```

Comment:
For my understanding, what would be the difference?

Reply:
I don't know! The update handler is passed a context but I don't know if it's different from the context passed to the runner. Probably not but it seems better to use it until I confirm it's okay

Reply:
I'm curious too - although I do think it's `d.ctx` being passed

Reply:
Confirmed with sdk team that update handlers must used the Context passed to them, not the Context of the top-level workflow. This is a pretty good argument for not putting the Context in the struct actually